system-contracts/contracts/precompiles/CodeOracle.yul

```diff
@@ -0,0 +1,134 @@
+/**
+ * @author Matter Labs
+ * @custom:security-contact security@matterlabs.dev
+ * @notice The contract used to emulate EVM's `extcodecopy` behavior.
+ */
+object "CodeOracle" {
+    code {
+        return(0, 0)
+    }
+    object "CodeOracle_deployed" {
+        code {
+            ////////////////////////////////////////////////////////////////
+            //                      CONSTANTS
+            ////////////////////////////////////////////////////////////////
+
+            /// @notice The fixed address of the known code storage contract.
+            function KNOWN_CODES_CONTRACT_ADDR() -> ret {
+                ret := 0x0000000000000000000000000000000000008004
+            }
+
+            ////////////////////////////////////////////////////////////////
+            //                      HELPER FUNCTIONS
+            ////////////////////////////////////////////////////////////////
+            
+            /// @notice The function that returns whether a certain versioned hash is marked as `known`
+            /// @param versionedHash The versioned hash to check
+            /// @return Whether the versioned hash is known
+            function isCodeHashKnown(versionedHash) -> ret {
+                // 1. Selector for `KnwonCodesStorage.getMarker(bytes32)`
+                mstore(0, 0x4c6314f000000000000000000000000000000000000000000000000000000000)
+                // 2. Input for `KnwonCodesStorage.getMarker(bytes32)`
+                mstore(4, versionedHash)
+
+                let success := staticcall(
+                    gas(),
+                    KNOWN_CODES_CONTRACT_ADDR(),
+                    0,
+                    36,
+                    0,
+                    32
+                )
+
+                if iszero(success) {
+                    // For some reason the call to the KnownCodesStorage failed.
+                    // Most likely out of gas.
+                    revert(0,0)
+                }
+
+                ret := mload(0)
+            }
+
+            /// @notice The cost for decommitment of a single 32-byte word.
+            function decommmitCostPerWord() -> ret {
+                ret := 4
+            }
+
+            /// @notice The function that performs that `decommit` operation, i.e. 
+            /// given a versioned hash (i.e. `commitment` to some blob), it unpacks it 
+            /// into the memory and returns it.
+            /// @param versionedHash The versioned hash to decommit.
+            /// @param lenInWords The length of the data in bytes to decommit.
+            function decommit(versionedHash, lenInWords) {
+                // The operation below are never expected to overflow since the `lenInWords` is a most 2 bytes long.
+                let gasCost := mul(decommmitCostPerWord(), lenInWords)
```

Comment:
Circuit reads only `uint32` value from there, so I would rather have a check that it is less than `type(uint32).max` too. 

I was trying to make complex attack with overflowing this value under `type(uint32).max` expecting that circuit will have a problem with that. I wasn't successful on making this attack, but in any case better check than sorry